packages/react/src/contexts/payment-routes.context.tsx

```diff
@@ -0,0 +1,169 @@
+import { ApiError } from '../definitions/error';
+import { useUserContext } from './user.context';
+
+interface PaymentRoutesInterface {
+  paymentRoutes?: PaymentRoutes;
+  paymentLinks?: PaymentLink[];
+  paymentRoutesLoading: boolean;
+  paymentLinksLoading: boolean;
+  createPaymentLink: (request: CreatePaymentLink) => Promise<void>;
+  updatePaymentLink: (request: UpdatePaymentLink, id?: number, externalId?: number) => Promise<void>;
+  createPaymentLinkPayment: (request: CreatePaymentLinkPayment, id?: number, externalId?: number) => Promise<void>;
+  cancelPaymentLinkPayment: (id?: number, externalId?: number) => Promise<void>;
+  error?: string;
+}
+
+const PaymentRoutesContext = createContext<PaymentRoutesInterface>(undefined as any);
+
+export function usePaymentRoutesContext(): PaymentRoutesInterface {
+  return useContext(PaymentRoutesContext);
+}
+
+export function PaymentRoutesContextProvider(props: PropsWithChildren): JSX.Element {
+  const { user } = useUserContext();
+  const {
+    getPaymentRoutes,
+    getPaymentLinks,
+    createPaymentLink: createPaymentLinkApi,
+    updatePaymentLink: updatePaymentLinkApi,
+    createPaymentLinkPayment: createPaymentLinkPaymentApi,
+    cancelPaymentLinkPayment: cancelPaymentLinkPaymentApi,
+  } = usePaymentRoutes();
+  const [error, setError] = useState<string | undefined>();
+  const [paymentRoutes, setPaymentRoutes] = useState<PaymentRoutes>();
+  const [paymentLinks, setPaymentLinks] = useState<PaymentLink[]>();
+  const [paymentRoutesLoading, setPaymentRoutesLoading] = useState<boolean>(false);
+  const [paymentLinksLoading, setPaymentLinksLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!user) {
+      setPaymentRoutes(undefined);
+      setPaymentLinks(undefined);
+      return;
+    }
+
+    // get routes
+    setPaymentRoutesLoading(true);
+    getPaymentRoutes()
+      .then((routes: PaymentRoutes) => {
+        routes.buy = routes.buy.filter((route) => route.active).sort(sortRoutes);
+        routes.sell = routes.sell.filter((route) => route.active).sort(sortRoutes);
+        routes.swap = routes.swap.filter((route) => route.active).sort(sortRoutes);
+        setPaymentRoutes(routes);
+      })
+      .catch((error: ApiError) => setError(error.message ?? 'Unknown error'))
+      .finally(() => setPaymentRoutesLoading(false));
+
+    // get payment links
+    setPaymentLinksLoading(true);
+    getPaymentLinks()
+      .then((links) => setPaymentLinks(links as PaymentLink[]))
+      .catch((error: ApiError) => setError(error.message ?? 'Unknown error'))
+      .finally(() => setPaymentLinksLoading(false));
+  }, [user]);
+
+  async function createPaymentLink(request: CreatePaymentLink): Promise<void> {
+    if (!user) return;
+
+    setPaymentLinksLoading(true);
+    createPaymentLinkApi(request)
+      .then(updatePaymentLinks)
+      .catch((error: ApiError) => setError(error.message ?? 'Unknown error'))
+      .finally(() => setPaymentLinksLoading(false));
```

Comment:
I think I wouldn't catch the error for change methods. Makes it much harder to handle them in the UI, if you want to.